### PR TITLE
Add stylus to the project...

### PR DESCRIPTION
**Pull request by darkwing**
*  Add stylus to the project
*  Adjusting run detail layout to two columns
*  Ensuring page stays the same upon comment form submission

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -13,3 +13,17 @@ This is a work in progress.  FOR DEMO ONLY.
 
 
 
+Development
+===========
+
+Setting the app up:
+
+	node install
```
Comment (pdehaan) [inline README.rst]: `npm install`?

I think `node install` will try and run ./install.js locally.

File: README.rst [hunk 2]
```diff
@@ -13,3 +13,17 @@ This is a work in progress.  FOR DEMO ONLY.
...
+
+Watch and compile stylus to CSS:
+
+	stylus --watch --compress client/static/assets/stylus/app.styl --out client/static/assets/css/
```
Comment (pdehaan) [inline README.rst]: Does this work, or would we need to install stylus locally/globally?

File: client/static/assets/stylus/app.styl [hunk 3]
```diff
@@ -1,3 +1,6 @@
+/*
+  Fonts, pulled from CDN for speed (global location for Mozilla fonts, used on Mozilla.org and MDN)
+*/
```
Comment (pdehaan) [inline client/static/assets/stylus/app.styl]: Sorry, I'm a Sass gentleman, but not sure if this is worth trying to slice out the `@font-face` declarations and try and include in app.styl as a @import or @require (if Stylus lets you do that stuffs).

But that's just a personal cosmetic thing. Feel free to ignore.

File: client/static/assets/views/_partials/comment-form.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <section>
-  <form role="form" name="commentForm" class="panel callout radius">
+  <form role="form" name="commentForm" class="panel callout radius" ng-action="{{ ROUTES.RUN_DETAIL }}/{{ runComment.runId }}" ng-submit="submitCommentForm();">
```
Comment (pdehaan) [inline client/static/assets/views/_partials/comment-form.html]: :+1:

File: README.rst (hunk 1 above)
Comment (darkwing) [inline README.rst]: Bleh, my bad, will fix.

File: README.rst (hunk 2 above)
Comment (pdehaan) [inline README.rst]: Actually, to self answer (sorry, pizza day at the office), I grabbed your branch and installed and was getting a white screen.
Turns out it was barfing on the lack of compiled CSS, so yes, we'll need to explicitly run this once before starting the server.
```sh
./node_modules/.bin/stylus --compress client/static/assets/stylus/app.styl --out client/static/assets/css/
```

A couple possible approaches:
1. Use grunt to compile CSS and possibly start server.
2. Create npm script to compile the CSS after doing `npm install` (similar to how we shim in the `bower install` command).
3. Create npm script to compile the CSS before doing `npm start` (doesn't work if you start the server manually, but would guarantee that the assets get generated before the server starts. May have slight downstream issues if you want to hack on the generated CSS manually for local dev.

I think this is probably as easy as creating a script in package.json `scripts` block which just calls:

```js
"stylus:compile": "stylus --compress client/static/assets/stylus/app.styl --out client/static/assets/css/"
```

And then adjust the `postinstall` or `prestart` scripts to do a `npm run stylus:compile`.

Comment (darkwing) [inline README.rst]: I added stylus to the `devDependencies` -- shouldn't that do the trick?

File: client/static/assets/stylus/app.styl (hunk 3 above)
Comment (darkwing) [inline client/static/assets/stylus/app.styl]: Fair point, I'll do this.

**Comment by pdehaan** [comment]
Looks good, apart from the Stylus questions above.

I did notice one bit of curious UI when you're looking at a details page in a narrower viewport (see screenshot below). Note how the breadcrumbs on the right cause the run id heading to overflow and wrap semi-awkwardly. Probably unrelated to this exact commit, but may be worthy of a separate PR.
nightly

**Comment by darkwing** [comment]
I noticed that as well, but was going to fix after this stylus PR went in so that the PR didn't get too big.  Will get on that once I have this set.

File: README.rst (hunk 2 above)
Comment (pdehaan) [inline README.rst]: Well, stylus will be installed, but you wouldn't probably be able to run it unless you installed it globally using `npm install stylus --global` (or `npm i stylus -g` for my fellow lazy typists).

You can run it manually using the `./node_modules/.bin/stylus` trick above, or if you're calling it via a script in package.json, that `./node_modules/.bin/` gets added to your path automatically.

I'm guessing you already have stylus installed globally which is why the `stylus` command works for you.

**Comment by darkwing** [comment]
OK, updated the README (stylus) and other issues.  I'll fix the breadcrumb issue in another PR!

**Comment by pdehaan** [comment]
Looks rad. Merging.

**Pull request merged**